### DISCUSSION TITLE
How do I output certain variables, currently being interpolated to atm and sfc netcdf files, to the UFS native grid?

### DISCUSSION THREAD

**Discussion by aschuh** [ufs-community/ufs-weather-model#1872] [Q&A] [answered]
Thanks to @dustinwales for the solution to where to find 3D physics variables added to diag_table in the output.  They WERE in the sfc.nc files.  Now, I'm outputting convective mass flux fields from the convective parameterization scheme and would like to get these output on the native grid in order to play around w/ tracer mixing external to the convection routine.  Interpolated fields make this difficult and the desire is to incorporate code back into UFS to do this, thus I need to test on columns on the native grid.  I've played around with turning quilting to FALSE in model_configure but seem to get errors/crashes on this, which look different based on values of *other* args in that file.  If ldiag3d = TRUE in input.nml, I can output det_mf, dwn_mf, upd_mf on interpolated grid using a model_configure that has quilting=TRUE.  How do I output these on the native tiles at an arbitrary time resolution (something close to model timestep, or at least convection timestep)?  

I'm working on ORION, using a RT test as a sandbox and my input.nml and model_configure are pasted below.

My input.nml:

&atmos_model_nml
  blocksize = 32
  chksum_debug = .false.
  dycore_only = .false.
  ccpp_suite = 'FV3_GFS_v17_p8_smoke_gsdchem'
/

&diag_manager_nml
  prepend_date = .false.
  max_output_fields = 300
  debug_diag_manager = .true.
  do_diag_field_log = .true.
/

&fms_io_nml
  checksum_required = .false.
  max_files_r = 100
  max_files_w = 100
/

&mpp_io_nml
shuffle=1
deflate_level=1
/

&fms_nml
  clock_grain = 'ROUTINE'
  domains_stack_size = 3000000
  print_memory_usage = .false.
/

&fv_core_nml
  layout = 3,8
  io_layout = 1,1
  npx = 97
  npy = 97
  ntiles = 6
  npz = 127
  dz_min = 6
  psm_bc = 1
  grid_type = -1
  make_nh = .true.
  fv_debug = .false.
  range_warn = .true.
  reset_eta = .false.
  n_sponge = 42
  nudge_qv = .true.
  nudge_dz = .false.
  tau = 10.0
  rf_cutoff = 7.5e2
  d2_bg_k1 = 0.20
  d2_bg_k2 = 0.04
  kord_tm = -9
  kord_mt = 9
  kord_wz = 9
  kord_tr = 9
  hydrostatic = .false.
  phys_hydrostatic = .false.
  use_hydro_pressure = .false.
  beta = 0.
  a_imp = 1.
  p_fac = 0.1
  k_split = 2
  n_split = 6
  nwat = 6
  na_init = 1
  d_ext = 0.
  dnats = 2
  fv_sg_adj = 450
  d2_bg = 0.
  nord = 2
  dddmp = 0.1
  d4_bg = 0.12
  vtdm4 = 0.02
  delt_max = 0.002
  ke_bg = 0.
  do_vort_damp = .true.
  external_ic = .true.
  external_eta = .true.
  gfs_phil = .false.
  nggps_ic = .true.
  mountain = .false.
  ncep_ic = .false.
  d_con = 1.
  hord_mt = 5
  hord_vt = 5
  hord_tm = 5
  hord_dp = -5
  hord_tr = 8
  adjust_dry_mass = .false.
  dry_mass=98320.0
  consv_te = 1.
  do_sat_adj = .false.
  consv_am = .false.
  fill = .true.
  dwind_2d = .false.
  print_freq = 6
  warm_start = .false.
  no_dycore = .false.
  z_tracer = .true.
  agrid_vel_rst = .true.
  read_increment = .false.
  res_latlon_dynamics = ''
/

&external_ic_nml
  filtered_terrain = .true.
  levp = 128
  gfs_dwinds = .true.
  checker_tr = .false.
  nt_checker = 0
/

&gfs_physics_nml
  fhzero       = 1
  h2o_phys     = .true.
  ldiag3d      = .true.
  qdiag3d      = .true.
  dtend_select = 'dtend*mp', 'dtend_temp_*' 
  naux2d         = 0
  naux3d         = 3
  aux3d_time_avg = .false.,.false.,.false.
  print_diff_pgr = .false.
  fhcyc        = 24
  use_ufo      = .true.
  pre_rad      = .false.
  imp_physics  = 8 
  iovr         = 3
  ltaerosol    = .false.
  mraerosol    = .true.
  lradar       = .false.
  ttendlim     = -999
  dt_inner     = 600
  sedi_semi    = .true.
  decfl        = 10
  oz_phys      = .false.
  oz_phys_2015 = .true.
  lsoil_lsm    = 4
  do_mynnedmf  = .false.
  do_mynnsfclay = .false.
  icloud_bl    = 1
  bl_mynn_edmf = 1
  bl_mynn_tkeadvect = .true.
  bl_mynn_edmf_mom = 1
  do_ugwp      = .false.
  do_tofd      = .false.
  gwd_opt      = 2
  do_ugwp_v0   = .true.
  do_ugwp_v1   = .false.
  do_ugwp_v0_orog_only = .false.
  do_ugwp_v0_nst_only  = .false.
  do_gsl_drag_ls_bl    = .false.
  do_gsl_drag_ss       = .true.
  do_gsl_drag_tofd     = .false.
  do_ugwp_v1_orog_only = .false.
  min_lakeice  = 0.15
  min_seaice   = 0.15
  use_cice_alb = .false.
  pdfcld       = .false.
  fhswr        = 3600.
  fhlwr        = 3600.
  ialb         = 2
  iems         = 2
  iaer         = 1011
  icliq_sw     = 2
  ico2         = 2
  isubc_sw     = 2
  isubc_lw     = 2
  isol         = 2
  lwhtr        = .true.
  swhtr        = .true.
  cnvgwd       = .true.
  shal_cnv     = .true.
  cal_pre      = .false.
  redrag       = .true.
  dspheat      = .true.
  hybedmf      = .false.
  satmedmf     = .true.
  isatmedmf    = 1
  lheatstrg    = .false.
  lseaspray    = .true.
  random_clds  = .false.
  trans_trac   = .true.
  cnvcld       = .true.
  imfshalcnv   = 2
  imfdeepcnv   = 2
  ras          = .false.
  cdmbgwd      = 0.14,1.8,1.0,1.0
  prslrd0      = 0.
  ivegsrc      = 1
  isot         = 1
  lsoil        = 4
  lsm          = 2
  iopt_dveg    = 4
  iopt_crs     = 2
  iopt_btr     = 1
  iopt_run     = 1
  iopt_sfc     = 3
  iopt_trs     = 2
  iopt_frz     = 1
  iopt_inf     = 1
  iopt_rad     = 3
  iopt_alb     = 1
  iopt_snf     = 4
  iopt_tbot    = 2
  iopt_stc     = 3
  debug        = .false.
  nstf_name    = 2,1,0,0,0
  nst_anl      = .true.
  psautco      = 0.0008,0.0005
  prautco      = 0.00015,0.00015
  lgfdlmprad   = .false.
  effr_in      = .true.
  ldiag_ugwp   = .false.
  fscav_aero   = "*:0.0","so2:0.0","msa:0.0","dms:0.0","nh3:0.4","nh4:0.6","bc1:0.6","bc2:0.6","oc1:0.4","oc2:0.4","dust1:0.6","dust2:0.6","dust3:0.6","dust4:0.6","dust5:0.6","seas1:0.5","seas2:0.5","seas3:0.5","seas4:0.5","seas5:0.5"
  do_sppt      = .false.
  do_shum      = .false.
  do_skeb      = .false.
  do_RRTMGP          = .false.
  active_gases       = 'h2o_co2_o3_n2o_ch4_o2'
  ngases             = 6
  lw_file_gas        = 'rrtmgp-data-lw-g128-210809.nc'
  lw_file_clouds     = 'rrtmgp-cloud-optics-coeffs-lw.nc'
  sw_file_gas        = 'rrtmgp-data-sw-g112-210809.nc'
  sw_file_clouds     = 'rrtmgp-cloud-optics-coeffs-sw.nc'
  rrtmgp_nGptsSW     = 112
  rrtmgp_nGptsLW     = 128
  rrtmgp_nBandsLW    = 16
  rrtmgp_nBandsSW    = 14
  doGP_cldoptics_LUT = .true.
  doGP_lwscat        = .true.
  use_med_flux = .false.
  frac_grid    = .true.
  cplchm       = .false.
  cplflx       = .false.
  cplice       = .false.
  cplwav       = .false.
  cplwav2atm   = .false.
  do_ca        = .true.
  ca_global    = .false.
  ca_sgs       = .true.
  nca          = 1
  ncells       = 5
  nlives       = 12
  nseed        = 1
  nfracseed    = 0.5
  nthresh      = 18
  ca_trigger   = .true.
  nspinup      = 1
  iseed_ca     = 12345
  lndp_type    = 0
  n_var_lndp   = 0
/

&cires_ugwp_nml
  knob_ugwp_solver  = 2
  knob_ugwp_source  = 1,1,0,0
  knob_ugwp_wvspec  = 1,25,25,25
  knob_ugwp_azdir   = 2,4,4,4
  knob_ugwp_stoch   = 0,0,0,0
  knob_ugwp_effac   = 1,1,1,1
  knob_ugwp_doaxyz  = 1
  knob_ugwp_doheat  = 1
  knob_ugwp_dokdis  = 1
  knob_ugwp_ndx4lh  = 1 
  knob_ugwp_version = 0
  launch_level = 54
/

&gfdl_cloud_microphysics_nml
  sedi_transport = .true.
  do_sedi_heat = .false.
  rad_snow = .true.
  rad_graupel = .true.
  rad_rain = .true.
  const_vi = .false.
  const_vs = .false.
  const_vg = .false.
  const_vr = .false.
  vi_max = 1.
  vs_max = 2.
  vg_max = 12.
  vr_max = 12.
  qi_lim = 1.
  prog_ccn = .false.
  do_qa = .true.
  fast_sat_adj = .true.
  tau_l2v = 225.
  tau_v2l = 150.
  tau_g2v = 900.
  rthresh = 10.e-6  ! This is a key parameter for cloud water
  dw_land  = 0.16
  dw_ocean = 0.10
  ql_gen = 1.0e-3
  ql_mlt = 1.0e-3
  qi0_crt = 8.0E-5
  qs0_crt = 1.0e-3
  tau_i2s = 1000.
  c_psaci = 0.05
  c_pgacs = 0.01
  rh_inc = 0.30
  rh_inr = 0.30
  rh_ins = 0.30
  ccn_l = 300.
  ccn_o = 100.
  c_paut = 0.5
  c_cracw = 0.8
  use_ppm = .false.
  use_ccn = .true.
  mono_prof = .true.
  z_slope_liq  = .true.
  z_slope_ice  = .true.
  de_ice = .false.
  fix_negative = .true.
  icloud_f = 1
  mp_time = 150.
  reiflag = 2
/

&interpolator_nml
  interp_method = 'conserve_great_circle'
/

&namsfc
  FNGLAC   = 'global_glacier.2x2.grb'
  FNMXIC   = 'global_maxice.2x2.grb'
  FNTSFC   = 'RTGSST.1982.2012.monthly.clim.grb'
  FNSNOC   = 'global_snoclim.1.875.grb'
  FNZORC   = 'igbp'
  FNALBC   = 'C96.snowfree_albedo.tileX.nc'
  FNALBC2  = 'C96.facsf.tileX.nc'
  FNAISC   = 'IMS-NIC.blended.ice.monthly.clim.grb'
  FNTG3C   = 'C96.substrate_temperature.tileX.nc'
  FNVEGC   = 'C96.vegetation_greenness.tileX.nc'
  FNVETC   = 'C96.vegetation_type.tileX.nc'
  FNSOTC   = 'C96.soil_type.tileX.nc'
  FNSMCC   = 'global_soilmgldas.statsgo.t1534.3072.1536.grb'
  FNMSKH   = 'global_slmask.t1534.3072.1536.grb'
  FNTSFA   = ''
  FNACNA   = ''
  FNSNOA   = ''
  FNVMNC   = 'C96.vegetation_greenness.tileX.nc'
  FNVMXC   = 'C96.vegetation_greenness.tileX.nc'
  FNSLPC   = 'C96.slope_type.tileX.nc'
  FNABSC   = 'C96.maximum_snow_albedo.tileX.nc'
  LDEBUG   =.false.
  FSMCL(2) = 99999
  FSMCL(3) = 99999
  FSMCL(4) = 99999
  LANDICE  = .false.
  FTSFS    = 90
  FAISL    = 99999
  FAISS    = 99999
  FSNOL    = 99999
  FSNOS    = 99999
  FSICL    = 0
  FSICS    = 0
  FTSFL    = 99999
  FVETL    = 99999
  FSOTL    = 99999
  FvmnL    = 99999
  FvmxL    = 99999
  FSLPL    = 99999
  FABSL    = 99999
/

&fv_grid_nml
  grid_file = 'INPUT/grid_spec.nc'
/

&nam_stochy
/

&nam_sfcperts
  lndp_type      = 0
  lndp_model_type = 0
  LNDP_TAU=21600,
  LNDP_LSCALE=500000,
  ISEED_LNDP=2010,
  lndp_var_list = 'XXX'
  lndp_prt_list = -999
/

&MOM_input_nml
  output_directory = 'MOM6_OUTPUT/',
  input_filename = 'n'
  restart_input_dir = 'INPUT/',
  restart_output_dir = 'RESTART/',
  parameter_filename = 'INPUT/MOM_input',
  'INPUT/MOM_override'/

model_configure:

start_year:              2021
start_month:             03
start_day:               22
start_hour:              06
start_minute:            0
start_second:            0
nhours_fcst:             3
fhrot:                   0

dt_atmos:                600
restart_interval:        1 -1
output_1st_tstep_rst:    .false.

quilting:                .true.
write_groups:            1
write_tasks_per_group:   6
itasks:                  1
output_history:          .true.
write_dopost:            .true.
write_nsflip:            .true.
num_files:               2
filename_base:           'atm' 'sfc'
output_grid:             gaussian_grid
output_file:             'netcdf'
ideflate:                0
nbits:                   0
ichunk3d:                0
jchunk3d:                0
kchunk3d:                0
imo:                     384
jmo:                     190

output_fh:               0 1 2 3
iau_offset:              0

**Comment by DeniseWorthen** [comment, marked as answer]
To get the cubed-sphere tiles, you'll need to change (in model_configure) the ``output_grid`` to cubed_sphere_grid and set ``write_dopost`` to .false.

To get the output at every ATM timestep, you can set ``output_fh: 0.167 -1`` 

0.167 = dt_atmos/3600 = 600/3600.

**Reply by aschuh** [reply]
Fantastic, thank you very much.

andrew
